Classes/Command/FluidTemplatePrecompileCommand.php

```diff
@@ -0,0 +1,163 @@
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Logger\ConsoleLogger;
+use Symfony\Component\Console\Output\Output;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use TYPO3Fluid\Fluid\Core\Cache\FluidCacheWarmupResult;
+
+/**
+ * A console command for precompiling Fluid templates
+ */
+class FluidTemplatePrecompileCommand extends Command
+{
+    use FluidPrecompilerServiceTrait;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setDescription('Precompile Fluid templates')
+            ->addArgument(
+                'extension',
+                InputArgument::OPTIONAL,
+                'Extension key of which the fluid template should be compiled, leave empty to compile all',
+                null
+            )
+            ->addOption(
+                'fail',
+                null,
+                InputOption::VALUE_NONE,
+                'Fail with a non-zero return code if any template could not be compiled'
+            )
+            ->addOption(
+                'show-only-failed',
+                null,
+                InputOption::VALUE_NONE,
+                'Show only failed fluid templates from compiling. Only in conjunction with -v'
+            )
+            ->addOption(
+                'limit',
+                'l',
+                InputOption::VALUE_REQUIRED,
+                'Limit of template files of each extension shown. Only in conjunction with -v',
+                5
+            )
+            ->addOption(
+                'disable-error-handler',
+                null,
+                InputOption::VALUE_NONE,
+                'Temporary disable any registered error handler for compiling'
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $outputHelper = new SymfonyStyle($input, $output);
+
+        // Get arguments and options
+        $extension = $input->getArgument('extension');
+        $fail = $input->getOption('fail');
+        $onlyFailed = $input->getOption('show-only-failed');
+        $limit = $input->getOption('limit');
+        $disableErrorHandlers = $input->getOption('disable-error-handler');
+
+        // Test exception handlers
+        if (set_error_handler(function() {}) && !$disableErrorHandlers) {
+            $outputHelper->warning(
+                "\nIt seems like there is an error handler registered."
+                . "\nIn some cases this can prevent the precompiling to run properly."
+                . "\nAn ErrorHandler might raise an exception for a PHP Warning."
+            );
+
+            restore_error_handler();
+        }
+
+        $result = $this->getPrecompilerService()->warmup($extension);
+
+        // General information
+        $templatesCount = 0;
+        foreach ($result as $extensionResult) {
```

Comment:
This could be done with `array_map()` and `array_sum()`

Reply:
What Mathias meant was `$templatesCount = array_sum(array_map('count', $result));` does the same. First it creates an array of counts of results for each extension, then sums those up.